webui/react/src/components/HyperparameterSearchModal.tsx

```diff
@@ -117,6 +117,12 @@ const HyperparameterSearchModal = ({ closeModal, experiment, trial }: Props): JS
     });
   }, [experiment.hyperparameters, trialHyperparameters]);
 
+  const [currentHPs, setCurrentHPs] =
+    useState<{ hyperparameter: Hyperparameter; name: string }[]>();
+  useEffect(() => {
+    !currentHPs && hyperparameters && setCurrentHPs(hyperparameters);
+  }, [hyperparameters, currentHPs]);
+
```

Comment:
can we use the form state for this instead?

Reply:
I might be wrong, but I don't think the state for this form is explicitly defined due to complexity